src/pull_request.ts

```diff
@@ -32,24 +36,39 @@ export async function handlePullRequest(config: Config) {
   const fileDiffs = files.map(parseFileDiff);
   info(`successfully fetched file diffs`);
 
-  // Create initial comment with the summary
-  const initialComment = await octokit.rest.issues.createComment({
-    ...context.repo,
-    issue_number: pull_request.number,
-    body: buildInitialMessage(
-      pull_request.base.sha,
-      pull_request.head.sha,
-      fileDiffs
-    ),
-  });
-  info(`posted initial comment`);
   // Get commit messages
   const { data: commits } = await octokit.rest.pulls.listCommits({
     ...context.repo,
     pull_number: pull_request.number,
   });
   info(`successfully fetched commit messages`);
 
+  // Find or create overview comment with the summary
+  const { data: comments } = await octokit.rest.issues.listComments({
+    ...context.repo,
+    issue_number: pull_request.number,
+  });
+  let overviewComment = comments.find((comment) =>
+    comment.body?.includes(OVERVIEW_MESSAGE_SIGNATURE)
+  );
```

Comment:
While you're using the optional chaining operator (`?.`), it's good practice to be explicit about null/undefined checks when searching for existing comments. Consider adding a null check:
```diff
- let overviewComment = comments.find((comment) =>
-   comment.body?.includes(OVERVIEW_MESSAGE_SIGNATURE)
- );
+ let overviewComment = comments.find((comment) =>
   comment.body && comment.body.includes(OVERVIEW_MESSAGE_SIGNATURE)
 );
```